module/VuFind/src/VuFind/AjaxHandler/GetItemStatuses.php

```diff
@@ -219,17 +213,48 @@ protected function reduceServices(array $rawServices)
     }
 
     /**
-     * Create a delimited version of the call number to allow the Javascript code
-     * to handle the prefix appropriately.
+     * Create an array with the callnumber and prefix of the given item.
      *
-     * @param string $prefix     Callnumber prefix or empty string.
-     * @param string $callnumber Main call number.
+     * @param array $item Item's holding data.
+     *
+     * @return array      Associative array with the keys 'prefix' and 'callnumber'
+     */
+    protected function getCallNumberArray(array $item): array
+    {
+        return [
+            'prefix' => $item['callnumber_prefix'] ?? '',
+            'callnumber' => $item['callnumber'],
+        ];
+    }
+
+    /**
+     * Render the callnumber HTML.
+     *
+     * @param string $callnumberSetting The callnumber mode setting
+     * @param array  $callnumbers       Callnumbers to render
      *
      * @return string
      */
-    protected function formatCallNo($prefix, $callnumber)
+    protected function renderCallnumbers(string $callnumberSetting, array $callnumbers): string
     {
-        return !empty($prefix) ? $prefix . '::::' . $callnumber : $callnumber;
+        $html = [];
+
+        $callnumberHandler = $this->getCallnumberHandler($callnumbers, $callnumberSetting);
+        foreach ($callnumbers as $number) {
+            $displayCallnumber = $actualCallnumber = $number['callnumber'];
+
+            if (!empty($number['prefix'])) {
+                $displayCallnumber = $number['prefix'] . ' ' . $displayCallnumber;
+            }
+
+            $html[] = $this->renderer->render('ajax/itemCallnumber', [
+                'actualCallnumber' => $actualCallnumber,
```

Comment:
You could use `compact('actualCallnumber', 'displayCallnumber', 'callnumberHandler')` to make this code a little simpler.